src/scripts/contracts/contractHelper.js

```diff
@@ -3,170 +3,217 @@ const axios = require('axios');
+	// console.log("unique", unique);
+	return unique;
 };
 
 const markPayableFunctions = (interfaceFunctions, uniqueFunctions) => {
-  const formattedObjects = [];
-  uniqueFunctions.forEach(element => {
-    const funcObj = {};
-    funcObj.signature = interfaceFunctions[element].signature;
-    funcObj.payable = interfaceFunctions[element]['payable'];
-    formattedObjects.push(funcObj);
-  });
-  return formattedObjects;
+	const formattedObjects = [];
+	uniqueFunctions.forEach(element => {
+		const funcObj = {};
+		funcObj.signature = interfaceFunctions[element].signature;
+		funcObj.payable = interfaceFunctions[element]['payable'];
+		formattedObjects.push(funcObj);
+	});
+	return formattedObjects;
 };
 
 const getFunctionInputs = (interfaceFunctions, formattedFunctions) => {
-  const functionsWithInputs = [];
-  const uniqueKey = 333;
-  for (let i = 0; i < formattedFunctions.length; i++) {
-    const functionObject = {};
-    const functionInputs = [];
-    for (var key in interfaceFunctions) {
-      if (key === formattedFunctions[i].signature) {
-        functionObject.signature = formattedFunctions[i].signature;
-        functionObject.payable = formattedFunctions[i].payable;
-        const inputs = interfaceFunctions[key].inputs;
-        for (let y = 0; y < inputs.names.length; y++) {
-          const inputObj = {};
-          inputObj.inputName = inputs.names[y];
-          inputObj.type = inputs.types[y];
-          const uniqueKeyHelper = functionsWithInputs.length * 2 + uniqueKey;
-          inputObj.uniqueKey = `${uniqueKeyHelper}${functionObject.signature}${inputObj.inputName}${functionObject.payable}`;
-          this.uniqueKey += 33;
-          functionInputs.push(inputObj);
-        }
-        functionObject.inputs = functionInputs;
-      }
-    }
-    functionsWithInputs.push(functionObject);
-  }
-  return functionsWithInputs;
+	const functionsWithInputs = [];
+	const uniqueKey = 333;
+	for (let i = 0; i < formattedFunctions.length; i++) {
+		const functionObject = {};
+		const functionInputs = [];
+		for (var key in interfaceFunctions) {
+			if (key === formattedFunctions[i].signature) {
+				functionObject.signature = formattedFunctions[i].signature;
+				functionObject.payable = formattedFunctions[i].payable;
+				const inputs = interfaceFunctions[key].inputs;
+				for (let y = 0; y < inputs.names.length; y++) {
+					const inputObj = {};
+					inputObj.inputName = inputs.names[y];
+					inputObj.type = inputs.types[y];
+					const uniqueKeyHelper = functionsWithInputs.length * 2 + uniqueKey;
+					inputObj.uniqueKey = `${uniqueKeyHelper}${functionObject.signature}${inputObj.inputName}${functionObject.payable}`;
+					this.uniqueKey += 33;
+					functionInputs.push(inputObj);
+				}
+				functionObject.inputs = functionInputs;
+			}
+		}
+		functionsWithInputs.push(functionObject);
+	}
+	return functionsWithInputs;
 };
 
 export const processContractByAddress = async (wallet, address, provider, network) => {
```

Comment:
Can we get a loading spinner to trigger on this. It has some agressive loops.

Reply:
@GregTheGreek a lot of the contract code can be simplified, I can clean this up a bit if you want

Reply:
v2 post cleanup

Reply:
@GregTheGreek @CFDevelopment  there are potential bugs in contracts functions, for example the **shuffles** function won't get correct inputs.

Reply:
Can you point this out in a comment ?

Reply:
I placed a review at the log where you can check the problem. @GregTheGreek